### PR TITLE
OSX CMake fixes for tests binary

### PR DESCRIPTION
**Pull request by KellenSunderland**
These changes allow the test projects to be properly built under OSX.
Some redundant sections in CMakeLists.txt have also been removed.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -152,7 +152,7 @@ include_directories("dlpack/include")
 #  add_subdirectory(dlpack)
 #endif()
 
-if(NOT MSVC)
+if(NOT MSVC AND NOT APPLE)
```
Comment (KellenSunderland) [inline CMakeLists.txt]: clang doesn't support this flag.  Static linking with default flags seems to work properly.

File: CMakeLists.txt [hunk 2]
```diff
@@ -218,14 +218,6 @@ else(USE_LAPACK)
   endif()
 endif()
 
-
```
Comment (KellenSunderland) [inline CMakeLists.txt]: This code looks like a c+p error.  Same import is above in this file.

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -39,7 +39,6 @@ if(GTEST_FOUND)
   if(UNITTEST_STATIC_LINK)
     target_link_libraries(${PROJECT_NAME}_unit_tests
       ${GTEST_LIBRARY}
-      rt
```
Comment (KellenSunderland) [inline tests/CMakeLists.txt]: mxnet already declares rt as a link dep.  The way in which mxnet is more platform independent than this mechanism, so I'd recommend we just let the mxnet targets declare this dep.

File: CMakeLists.txt (hunk 2 above)
Comment (cjolivier01) [reply CMakeLists.txt]: I didn't understand this comment?

File: tests/CMakeLists.txt (hunk 3 above)
Comment (cjolivier01) [reply tests/CMakeLists.txt]: Declares it into which macro? Is that macro used here?  Is it added into the macronin such a way that it's always part of the macro at this point?

File: CMakeLists.txt (hunk 2 above)
Comment (KellenSunderland) [reply CMakeLists.txt]: This block https://github.com/apache/incubator-mxnet/blob/master/CMakeLists.txt#L160 seems to be a copy and paste of this block https://github.com/apache/incubator-mxnet/blob/master/CMakeLists.txt#L222 .  Does having it twice serve some purpose, or are we just appending it to the same list twice?

File: tests/CMakeLists.txt (hunk 3 above)
Comment (KellenSunderland) [reply tests/CMakeLists.txt]: I should make it clear that I'm not a cmake expert, but to me it looks like it'll be included in mxnet_LINKER_LIBS from line https://github.com/apache/incubator-mxnet/blob/master/CMakeLists.txt#L225 when required (i.e. in Unix, not apple).  and mxnet_LINKER_LIBS is referenced both here https://github.com/apache/incubator-mxnet/blob/master/tests/CMakeLists.txt#L45 and here https://github.com/apache/incubator-mxnet/blob/master/tests/CMakeLists.txt#L54 in the same scope as rt.

**Comment by cjolivier01** [comment]
Hi can you rebase from master and push to kick off a new build, please? Hopefully fixes that unit test.

**Comment by oza** [comment]
The reason of test failure looks to be related to #8087. Please don't commit this before passing all tests.

**Comment by KellenSunderland** [comment]
I think I'm missing how it's related.  It looks like to me the latest failure was a memory corruption error.
```bash
test_ndarray.test_cached ... *** Error in `/usr/bin/python3': malloc(): smallbin double linked list corrupted: 0x00007fad500008c0 ***
script returned exit code 134
```
No rush to merge this one though.  I'll periodically rebase it until the tests pass.

**Comment by piiswrong** [comment]
Is this ready for merging?

**Comment by cjolivier01** [comment]
Yes

**Comment by cjolivier01** [comment]
Assuming CI can pass

**Pull request merged**